api/v1alpha1/clustergroupupgrade_types.go

```diff
@@ -113,8 +113,29 @@ type ClusterGroupUpgradeSpec struct {
 	//   - label1Name
 	// All the clusters matching the labels specified in clusterSelector will be included
 	// in the update plan.
+	// Deprecated: Use ClusterLabelSelectors instead
```

Comment:
This is just a comment, right? Is there a markup for kubebuilder so the API is generated with some deprecated annotation?

Reply:
I've added a kubebuilder annotation for deprevated version that I found in the docs here: https://book.kubebuilder.io/reference/markers/crd.html

Reply:
:+1: , how did I miss the very next line ...